daemon/mgr/cri_test.go

```diff
@@ -99,6 +172,37 @@ func TestToCriSandboxState(t *testing.T) {
 
 // Container related unit tests.
 
+func TestParseContainerName(t *testing.T) {
+	format := fmt.Sprintf("%s_${container name}_${sandbox name}_${sandbox namespace}_${sandbox uid}_${attempt times}", kubePrefix)
+
+	longerNameContainer := "k8s_cname_name_namespace_uid_3_4"
+	expectedError := "failed to parse container name: ," + longerNameContainer + "which should be" + format
+	_, LongerError := parseContainerName(longerNameContainer)
+	assert.EqualError(t, LongerError, expectedError)
+
+	WrongPrefixContainer := "Swarm_cname_name_namespace_uid_3"
+	expectedError = "container is not managed by kubernetes: " + WrongPrefixContainer
+	_, WrongPrefixError := parseContainerName(WrongPrefixContainer)
+	assert.EqualError(t, WrongPrefixError, expectedError)
+
+	WrongAttemptContainer := "k8s_cname_name_namespace_uid_notInt"
+	parts := strings.Split(WrongAttemptContainer, nameDelimiter)
+	_, err := parseUint32(parts[5])
+	expectedError = fmt.Sprintf("failed to parse the attempt times in container name: %q: %v", WrongAttemptContainer, err)
+	_, WrongAttemptError := parseContainerName(WrongAttemptContainer)
+	assert.EqualError(t, WrongAttemptError, expectedError)
+
+	SuccessParseContainer := "k8s_cname_name_namespace_uid_3"
```

Comment:
Successful case may not be needed, it will be tested in `TestContainerNameRoundTrip`.

Reply:
Should I delete this successful case?